### PR TITLE
Go Client HTTP header and query params support

### PR DESCRIPTION
**Pull request by weilueluo**
https://jira-eng-rtp3.cisco.com/jira/browse/LH-67704

### Description
- Allow manipulation of request header and query params before sending it.
- Also added missing cloudfmc read tests
- Renamed `cdfmc` -> `cloudfmc` as it is more clear to external user
- Fix failing tests because of adding header & query params support

### CODE REVIEW

File: client/device/cloudfmc/readaccesspolicies.go [hunk 1]
```diff
@@ -1,31 +1,38 @@
...
 func ReadAccessPolicies(ctx context.Context, client http.Client, inp ReadAccessPoliciesInput) (*ReadAccessPoliciesOutput, error) {
 
-	readUrl := url.ReadAccessPolicies(client.BaseUrl(), inp.DomainUid, inp.Limit)
+	client.Logger.Println("reading FMC Access Policies")
+
+	readUrl := url.ReadAccessPolicies(client.BaseUrl(), inp.DomainUid)
 
 	req := client.NewGet(ctx, readUrl)
+	req.Header.Add("Fmc-Hostname", inp.FmcHostname) // required, otherwise 500 internal server error
+	req.QueryParams.Add("limit", strconv.Itoa(inp.Limit))
```
Comment (weilueluo) [inline client/device/cloudfmc/readaccesspolicies.go:35]: example usage here

File: client/internal/http/request.go [hunk 2]
```diff
@@ -46,6 +48,9 @@ func NewRequest(config cdo.Config, httpClient *http.Client, logger *log.Logger,
 		method: method,
 		url:    url,
 		body:   body,
+
+		Header:      make(http.Header),
```
Comment (siddhuwarrier) [inline client/internal/http/request.go:52]: Why don't you make this an array, so you can pass in multiple headers? Why restrict to just one?

Comment (weilueluo) [reply client/internal/http/request.go:52]: Actually, it does support multiple headers, the name that golang chose is quite confusing, but `http.Header` is actually a `map[string][]string` (map of string to string array), it has a method call `Add` which is actually an `append` internal to the string array. So it is more similar to [Guava's multimap](https://guava.dev/releases/23.0/api/docs/com/google/common/collect/Multimap.html) interface

File: client/internal/http/request.go [hunk 3]
```diff
@@ -46,6 +48,9 @@ func NewRequest(config cdo.Config, httpClient *http.Client, logger *log.Logger,
 		method: method,
 		url:    url,
 		body:   body,
+
```
[the remark this answers is not in the document]
Comment (weilueluo) [reply client/internal/http/request.go:51]: Yes that would be more explicit, if we want to generalise this, then people will start adding all kinds of different headers to this struct, or even different body types, but I agree content type json is very commonly used, I have added specialised method to append that header if not given.

**Review by siddhuwarrier — APPROVED** [review]

**Pull request merged**